### PR TITLE
[NavBar] Add notifications badge and auth

### PR DESCRIPTION
**Pull request by damassi**
This PR abstracts the notification system slightly by moving the `QueryRenderer` out of the `<NotificationsMenu>` proper so that it can be shared by a new NotificationsBadge. It also wires up basic auth via `mediator` (depends on https://github.com/artsy/force/pull/3701). 

**The flow**: 
- Page loads and a request is kicked off for notifications
- On complete, the badge updates. This simultaneously warms `react-relay-network-layer`'s cache
- When a user hovers over the Notification badge and a drop down appears sub-items are instantly available, since both the menu and the badge share the same query / query renderer.
- A cookie is set with the notification value. During the SSR render pass (or when notifications are fetching) the badge will attempt to read from the cookie, which is passed over via Sharify. 

#### Todo:
- Tests. Going to start wiring those up in a follow-up PR. 

badge

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit c92d7adcaa9ef509ae88af84c0ee024db965392c

https://deploy-preview-2133--artsy-reaction.netlify.com

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit b6ff5c5b16f2b301a3e4bd891219597c5dfb1031

https://deploy-preview-2133--artsy-reaction.netlify.com

File: src/Components/NavBar/Menus/NotificationsMenu.tsx [hunk 1]
```diff
@@ -72,43 +77,53 @@ const NotificationsMenu: React.FC<NotificationsMenuQueryResponse> = props => {
...
-                        }
+    <QueryRenderer<NotificationsMenuQuery>
+      environment={relayEnvironment}
+      query={graphql`
+        query NotificationsMenuQuery {
+          me {
+            followsAndSaves {
+              notifications: bundledArtworksByArtist(
+                sort: PUBLISHED_AT_DESC
+                for_sale: true
```
Comment (damassi) [inline src/Components/NavBar/Menus/NotificationsMenu.tsx]: @mzikherman - does this flag mean _only_ for sale items, or is it inclusive?

Comment (damassi) [reply src/Components/NavBar/Menus/NotificationsMenu.tsx]: @mzikherman - ping

Comment (mzikherman) [reply src/Components/NavBar/Menus/NotificationsMenu.tsx]: Yea, this param means only for sale items.

**Comment by damassi** [comment]
Alright, this is ready for a look @mzikherman.

**Comment by damassi** [comment]
Merging per offline convo with @mzikherman

**Comment by artsyit** [comment]
:rocket: PR was released in v16.6.5 :rocket:

**Pull request merged**